ci_tests/src/index.ts

```diff
@@ -137,156 +205,284 @@ async function deployEth(
 
   // initialize() on both the manager and transceiver
   console.log("Initialize the manager");
-  await (await manager.initialize()).wait();
+  await tryAndWaitThrice(() => manager.initialize());
   console.log("Initialize the transceiver");
-  await (await transceiver.initialize()).wait();
+  await tryAndWaitThrice(() => transceiver.initialize());
 
   // Setup the initial calls, like transceivers for the manager
   console.log("Set transceiver for manager");
-  await (await manager.setTransceiver(await transceiver.address)).wait();
+  await tryAndWaitThrice(() => manager.setTransceiver(transceiver.address));
 
   console.log("Set outbound limit for manager");
-  await (await manager.setOutboundLimit(utils.parseEther("10000"))).wait();
+  await tryAndWaitThrice(() =>
+    manager.setOutboundLimit(utils.parseEther("10000"))
+  );
 
   return {
+    type: "evm",
     chainId: chain_id,
-    transceiverAddress: await transceiverProxyAddress.address,
-    managerAddress: await managerProxyAddress.address,
+    chainName,
+    transceiverAddress: transceiverProxyAddress.address,
+    managerAddress: managerProxyAddress.address,
     NTTTokenAddress: ERC20NTTAddress,
     wormholeCoreAddress: "0xC89Ce4735882C9F0f0FE26686c53074E09B0D550", // Same on both of the chains
     signer,
   };
 }
 
-async function link(chain1: ChainDetails, chain2: ChainDetails) {
+async function initSolana(): Promise<SolanaChainDetails> {
+  console.log("Using public key", SOL_PUBLIC_KEY.toString());
+  const mint = await spl.createMint(
+    SOL_CONNECTION,
+    SOL_PRIVATE_KEY,
+    SOL_PUBLIC_KEY,
+    null,
+    9
+  );
+  console.log("Created mint", mint.toString());
+  const tokenAccount = await spl.createAssociatedTokenAccount(
+    SOL_CONNECTION,
+    SOL_PRIVATE_KEY,
+    mint,
+    SOL_PUBLIC_KEY
+  );
+  console.log("Created token account", tokenAccount.toString());
+  // await spl.mintTo(
```

Comment:
it will need to be done for the locking hub, which is why I left it (was planning to add a parameter to this function that would optionally do the `mintTo` and toggle the manager to `locking`)